plugins/text-editor-resources/src/components/extension/inlineComment.ts

```diff
@@ -0,0 +1,984 @@
+  destructors: Array<() => void>
+
+  handlers: {
+    handleResolveThread: () => void
+    handleSubmit: (text: string, _id?: string) => void
+  } | null
+
+  constructor (pluginOptions: InlineCommentExtensionOptions, props: InlineCommentViewProps) {
+    this.pluginOptions = pluginOptions
+    this.props = props
+    this.container = null
+    this.renderer = null
+    this.tippynode = null
+    this.destructors = []
+    this.handlers = null
+
+    this.rect = {
+      x: 0,
+      y: 0,
+      width: 0,
+      height: 0
+    }
+  }
+
+  buildComponentProps (view: EditorView): ThreadPresenterProps {
+    return {
+      thread: this.props.thread,
+      autofocus: this.props.autofocus ?? false,
+      width: commentWidth,
+      highlighted: this.props.highlighted,
+      collapsed: this.props.collapsed ?? false,
+      handleResolveThread: this.handlers?.handleResolveThread,
+      handleSubmit: this.handlers?.handleSubmit
+    }
+  }
+
+  build (view: EditorView, props?: InlineCommentViewProps): HTMLElement {
+    const prevprops = this.props
+    if (props !== undefined) this.props = props
+
+    if (this.handlers === null) {
+      this.handlers = {
+        handleResolveThread: () => {
+          resolveThread(this.pluginOptions, view, this.props.thread._id)
+        },
+        handleSubmit: (text: string, _id?: string) => {
+          updateThreadComment(this.pluginOptions, view, { _id, thread: this.props.thread._id, message: text })
+        }
+      }
+    }
+
+    const displayMode = this.props.displayMode
+    const placement = displayMode === 'compact' ? 'bottom' : 'right-start'
+
+    const getReferenceClientRect = (): DOMRect => {
+      return displayMode === 'full'
+        ? getReferenceAnchor(view, this.props.range.from, this.props.range.to)
+        : getReferenceRect(view, this.props.range.from, this.props.range.to)
+    }
+
+    if (this.tippynode === null && this.props.component !== undefined) {
+      this.container = document.createElement('div')
+
+      const listeners: Array<[keyof HTMLElementEventMap, () => void]> = [
+        [
+          'click',
+          (): void => {
+            updatePointerState(view, { focus: new Set([this.props.thread._id]) })
+          }
+        ],
+        [
+          'mouseenter',
+          (): void => {
+            updatePointerState(view, { hover: new Set([this.props.thread._id]) })
+          }
+        ],
+        [
+          'mouseleave',
+          (): void => {
+            updatePointerState(view, { hover: new Set([]) })
+          }
+        ]
+      ]
+
+      for (const listener of listeners) {
+        this.container.addEventListener(listener[0], listener[1])
+        this.destructors.push(() => this.container?.removeEventListener(listener[0], listener[1]))
+      }
+
+      this.destructors.push(
+        observeSizeChanges(this.container, () => {
+          forceUpdateDecorations(view)
+        })
+      )
+
+      this.tippynode = tippy(view.dom, {
```

Comment:
can we do this without tippy on our tooltips only?

Reply:
It would be easier to create a new tooltip system from scratch. Unfortunately the current system can't help in any meaningful way, whereas popperjs and tippy are shipped with tiptap and already used for the editor menus.